frontends/election-manager/src/screens/tally_writein_report_screen.tsx

```diff
@@ -0,0 +1,351 @@
+  const { votingMethod: votingMethodFromProps } =
+    useParams<VotingMethodReportScreenProps>();
+  const votingMethod = votingMethodFromProps as VotingMethod;
+  const {
+    electionDefinition,
+    isOfficialResults,
+    fullElectionTally,
+    isTabulationRunning,
+    auth,
+    logger,
+  } = useContext(AppContext);
+  assert(electionDefinition);
+  assert(isElectionManagerAuth(auth)); // TODO(auth) check permissions for viewing tally reports.
+  const userRole = auth.user.role;
+
+  const { election } = electionDefinition;
+  const statusPrefix = isOfficialResults ? 'Official' : 'Unofficial';
+
+  useEffect(() => {
+    if (previewReportRef?.current && printReportRef?.current) {
+      previewReportRef.current.innerHTML = printReportRef.current.innerHTML;
+    }
+  }, [previewReportRef, printReportRef, writeInCounts]);
+
+  // Get write-in tallies
+  useEffect(() => {
+    async function fetchAdjudicationCounts() {
+      const res = await fetch('/admin/write-ins/adjudications/counts');
+      const map = new Map<ContestId, Map<string, number>>();
+      for (const {
+        contestId,
+        transcribedValue,
+        adjudicationCount,
+      } of await res.json()) {
+        const innerMap = map.get(contestId) || new Map<string, number>();
+        innerMap.set(transcribedValue, adjudicationCount);
+        map.set(contestId, innerMap);
+      }
+      setWriteInCounts(map);
+    }
+    void fetchAdjudicationCounts();
+  }, [setWriteInCounts]);
+
+  const precinctName =
+    (precinctId &&
+      precinctId !== 'all' &&
+      find(election.precincts, (p) => p.id === precinctId).name) ||
+    undefined;
+
+  const fileSuffix = useMemo(() => {
+    if (scannerId) {
+      return `scanner-${scannerId}`;
+    }
+    if (batchId) {
+      return `batch-${batchId}`;
+    }
+    if (precinctId === 'all') {
+      return 'all-precincts';
+    }
+    if (partyId) {
+      const party = find(election.parties, (p) => p.id === partyId);
+      return party.fullName;
+    }
+    if (votingMethod) {
+      const label = getLabelForVotingMethod(votingMethod);
+      return `${label}-ballots`;
+    }
+    return precinctName;
+  }, [
+    batchId,
+    scannerId,
+    precinctId,
+    votingMethod,
+    partyId,
+    election.parties,
+    precinctName,
+  ]);
+  const batchLabel = useMemo(() => {
+    if (batchId) {
+      assert(fullElectionTally);
+      const batchTally = filterTalliesByParamsAndBatchId(
+        fullElectionTally,
+        election,
+        batchId,
+        {}
+      );
+      return `${batchTally.batchLabel} (Scanner: ${batchTally.scannerIds.join(
+        ', '
+      )})`;
+    }
+    return '';
+  }, [batchId, fullElectionTally, election]);
+  const reportDisplayTitle = useMemo(() => {
+    if (precinctName) {
+      return `${statusPrefix} Precinct Write-In Tally Report for ${precinctName}`;
+    }
+    if (scannerId) {
+      return `${statusPrefix} Scanner Write-In Tally Report for Scanner ${scannerId}`;
```

Comment:
We're not actually supporting this broken down in all of these ways right so we can probably remove some of this? (I know you just copied this)

Reply:
Will we ever want this?

Reply:
yeah idk, I had started to remove it in my PR on top of this and then had the same thought, I'm not sure it really will make sense ever to break down in those ways? But we can leave for now since its simpler at this point to leave it haha

Reply:
I'm guessing we will want by party eventually but not the others